providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/GrpcConnector.java

```diff
@@ -89,14 +89,16 @@ public void shutdown() throws Exception {
         }
 
         try {
-            if (this.channel != null) {
+            if (this.channel != null && !this.channel.isShutdown()) {
                 this.channel.shutdown();
-                this.channel.awaitTermination(5, TimeUnit.SECONDS);
+                this.channel.awaitTermination(this.deadline, TimeUnit.MILLISECONDS);
             }
         } finally {
             this.cache.clear();
-            if (this.channel != null) {
+            if (this.channel != null && !this.channel.isShutdown()) {
                 this.channel.shutdownNow();
+                this.channel.awaitTermination(5000, TimeUnit.MILLISECONDS);
```

Comment:
do we need both here 🤔 

shutdown now - Initiates a forceful shutdown in which preexisting and new calls are cancelled 
awaitTermination - Waits for the channel to become terminated, giving up if the timeout is reached 

I think we do not need both here. But rather, Log warning -> call `shutdownNow()`. 🤔

Reply:
From the javadoc:

```
initiates a forceful shutdown in which preexisting and new calls are cancelled. Although forceful, **_the shutdown process is still not instantaneous; [isTerminated()](vscode-file://vscode-app/usr/lib/code/out/vs/code/electron-sandbox/workbench/workbench.html) will likely return false immediately after this method returns._**
```

Reply:
This should be using `thid.deadline` though, not `5000`. I will fix that.